tests/functional/backend/corpora/test_revisions.py

```diff
@@ -0,0 +1,291 @@
+            self.assertEqual(len(public_datasets), 2)
+            ids = [dataset["id"] for dataset in public_datasets]
+            self.assertIn(another_dataset_id, ids)
+
+        # deleted_dataset_id = None
+        with self.subTest("Deleting a dataset does not effect the published dataset"):
+            # Start a revision
+            res = requests.post(f"{self.api}/dp/v1/collections/{collection_uuid}", headers=headers)
+            self.assertEqual(res.status_code, 201)
+
+            # Pick a random dataset to delete
+            dataset_to_delete = res.json()['datasets'][0]
+            deleted_dataset_id = dataset_to_delete['id']
+            original_dataset_id = dataset_to_delete['original_id']
+            original_explorer_url = self.create_explorer_url(original_dataset_id)
+
+            # Delete a dataset within the revision
+            res = requests.delete(f"{self.api}/dp/v1/datasets/{deleted_dataset_id}", headers=headers)
+            self.assertEqual(res.status_code, 202)
+
+            # Check if the dataset is still available
+            res = requests.get(f"{self.api}/dp/v1/datasets/meta?url={original_explorer_url}")
+            self.assertEqual(res.status_code, 200)
+
+            # TODO: This fails
+            res = requests.get(f"{self.api}/cellxgene/e/{original_dataset_id}/api/v0.2/schema")
+            self.assertEqual(res.status_code, 200)
+
+        with self.subTest("Publishing a revision that deletes a dataset removes it from the data portal"):
+            # Publish the revision
+            res = requests.post(f"{self.api}/dp/v1/collections/{collection_uuid}/publish", headers=headers)
+            res.raise_for_status()
+            self.assertEqual(res.status_code, requests.codes.accepted)
+
+            # Check that the dataset doesn't exist anymore
+            res = requests.get(f"{self.api}/dp/v1/datasets/meta?url={self.create_explorer_url(deleted_dataset_id)}")
+            self.assertEqual(res.status_code, 404)
+
+            # TODO: should also check original_dataset_id, but it returns 200
+
+            # TODO: add `And the explorer url redirects appropriately`
+
+
+            # Publishing a revised dataset replaces the original dataset
+            #     Check the metadata api to ensure the s3_uri has changed
+            #     Check the the explorer/schema endpoint to ensure the explorer is accessing the correct dataset
+            # Deleting a dataset does not effect the published dataset
+            # Publishing a revision that deletes a dataset removes it from the data portal (And the explorer url redirects appropriately)
+            # Adding a dataset to a revision does not impact public datasets in that collection
+            # Publishing a revision that contains a new dataset updates the collections page for the data portal (with the new dataset)
+
+
+    def upload_and_wait(self, collection_uuid, dropbox_url, existing_dataset_id = None):
```

Comment:
It might be worth pulling this and some of the other helper functions out of this file and the test_api file and storing them on a base class for functional tests

Reply:
Definitely, I can work on this.